solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/JDBCStream.java

```diff
@@ -275,12 +277,10 @@ public void setStreamContext(StreamContext context) {
     this.streamContext = context;
   }
 
-  /** Opens the JDBCStream */
-  public void open() throws IOException {
-
+  protected Driver getDriver() throws IOException {
     try {
       if (null != driverClassName) {
-        Class.forName(driverClassName);
+        Class.forName(driverClassName, true, getClass().getClassLoader());
```

Comment:
Should probably add a comment here making it clear there is intention in using this classloader and what that intention is.

I'm a little unclear the intention myself. I belive Class.forName(className) actually calls Class.forName(className, true, this.getClass().getClassLoader()?

Generally you want to avoid both if at all possible. Not only do they potentially behave differently depending on if called from a static context or not, but you pretty much always want to strive for getting the context classloader off the current thread because it will generally at least try and keep the classpath hiearchy sensible - by using a class classloader, there is always high potential that you get instantiated by a classloader higher up the chain than what is used by other application code, and the classes from that code will then be invisible to you.

Reply:
So `Class.forName(className)` actually uses `ClassLoader.getClassLoader(ClassLoader.getClassLoader(caller))`, which apparently does not work the same as `getClass().getClassLoader()`.

Unfortunately we can't use SolrResourceLoader here, because streams are in solrj for some reason and SolrResourceLoader is unsurprisingly in core. But agree that now that we better control thread classloaders, we should use that instead.

Reply:
Actually none of this matters. later on `DriverManager.getDriver(connectionUrl)` will check for the driver class on the JDBCStream.class's classLoader. This will fail as well, so no real reason to protect against this for now. We'd have to really get rid of the DriverManager all-together to remove ourselves from this error case.

I'm going to go ahead and just use `Class.forName(className)`

Reply:
Good point, I'll add a section explaining the downsides, but the reasoning as to why it is the way it is.